lib/ansible/modules/windows/win_shortcut.ps1

```diff
@@ -0,0 +1,147 @@
+#!powershell
+# (c) 2016, Dag Wieers <dag@wieers.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+#
+# Based on: http://powershellblogger.com/2016/01/create-shortcuts-lnk-or-url-files-with-powershell/
+
+$ErrorActionPreference = "Stop"
+
+$params = Parse-Args $args -supports_check_mode $true;
+
+$_ansible_check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -default $false
+
+$orig_src = Get-AnsibleParam -obj $params -name "src" -default $null
+$orig_dest = Get-AnsibleParam -obj $params -name "dest" -failifempty $true
+$state = Get-AnsibleParam -obj $params -name "state" -default "present"
+$orig_args = Get-AnsibleParam -obj $params -name "args" -default $null
+$orig_directory = Get-AnsibleParam -obj $params -name "directory" -default $null
+$hotkey = Get-AnsibleParam -obj $params -name "hotkey" -default $null
+$orig_icon = Get-AnsibleParam -obj $params -name "icon" -default $null
+$orig_description = Get-AnsibleParam -obj $params -name "description" -default $null
+$windowstyle = Get-AnsibleParam -obj $params -name "windowstyle" -default $null
+
+# Expand environment variables (Beware: turns $null into "")
+$src = [System.Environment]::ExpandEnvironmentVariables($orig_src)
+$dest = [System.Environment]::ExpandEnvironmentVariables($orig_dest)
+$args = [System.Environment]::ExpandEnvironmentVariables($orig_args)
+$directory = [System.Environment]::ExpandEnvironmentVariables($orig_directory)
+$icon = [System.Environment]::ExpandEnvironmentVariables($orig_icon)
+$description = [System.Environment]::ExpandEnvironmentVariables($orig_description)
+
+$result = New-Object PSObject @{
```

Comment:
Still interested in your view on this. Because I did wonder why it was a PSObject (whatever that is).
I prefer basic datatypes if possible, so I would like to rewrite it myself, but other modules do this.